csrc/ir/utils.cpp

```diff
@@ -1414,15 +1418,21 @@ bool matchMatmulPatterns(const UnaryOp* cast_op, MatmulInputs* matmul_inp) {
 
   // Verify assumptions for MmaOp hold. Assign the values to Mma operands.
   MmaOpUtils::verifyMmaOpForEvaluation(mma, final_out_dtype);
-  matmul_inp->mma_lhs = mma->inA();
-  matmul_inp->mma_rhs = mma->inB();
+
+  // Get the non-broadcasted values to avoid inferring squeeze dimensions.
+  matmul_inp->mma_lhs = mma->inA()->definition()->input(0);
+  matmul_inp->mma_rhs = mma->inB()->definition()->input(0);
+  matmul_inp->input_layout =
+      mma->inA()->as<TensorView>()->getRootDomain().back()->isBroadcast()
```

Comment:
Ha, I see what you mean by that now.

Reply:
It was to avoid this issue:https://github.com/NVIDIA/Fuser/issues/2057#issuecomment-2046133253
We also discussed this case offline.